saleor/graphql/checkout/dataloaders.py

```diff
@@ -26,54 +26,65 @@ class CheckoutLinesInfoByCheckoutTokenLoader(DataLoader):
     context_key = "checkoutlinesinfo_by_checkout"
 
     def batch_load(self, keys):
+        from ..channel.dataloaders import ChannelByCheckoutLineIDLoader
+
         def with_checkout_lines(checkout_lines):
             variants_pks = list(
                 {line.variant_id for lines in checkout_lines for line in lines}
             )
             if not variants_pks:
                 return [[] for _ in keys]
-
-            def with_variants_products_collections(results):
-                variants, products, collections, channel_listings = results
-                variants_map = dict(zip(variants_pks, variants))
-                products_map = dict(zip(variants_pks, products))
-                collections_map = dict(zip(variants_pks, collections))
-                channel_listings_map = dict(zip(variants_pks, channel_listings))
-
-                lines_info = []
-                for lines in checkout_lines:
-                    lines_info.append(
-                        [
-                            CheckoutLineInfo(
-                                line=line,
-                                variant=variants_map[line.variant_id],
-                                channel_listing=channel_listings_map[line.variant_id],
-                                product=products_map[line.variant_id],
-                                collections=collections_map[line.variant_id],
-                            )
-                            for line in lines
-                        ]
-                    )
-                return lines_info
-
-            variants = ProductVariantByIdLoader(self.context).load_many(variants_pks)
-            products = ProductByVariantIdLoader(self.context).load_many(variants_pks)
-            collections = CollectionsByVariantIdLoader(self.context).load_many(
-                variants_pks
+            channel = ChannelByCheckoutLineIDLoader(self.context).load(
+                checkout_lines[0][0].id
```

Comment:
What if `checkout_lines` is an empty list?

Reply:
`variants_pks` will be empty then and the function will return in line 36.